src/controllers/blocks/BlocksController.ts

```diff
@@ -233,4 +236,71 @@ export default class BlocksController extends AbstractController<BlocksService>
 			await this.service.fetchBlockHeader(hash)
 		);
 	};
+
+	/**
+	 * Return a collection of blocks, given a range.
+	 *
+	 * @param req Express Request
+	 * @param res Express Response
+	 */
+	private getBlocks: RequestHandler<
+		unknown,
+		unknown,
+		unknown,
+		IRangeQueryParam
+	> = async (
+		{ query: { range, eventDocs, extrinsicDocs } },
+		res
+	): Promise<void> => {
+		if (!range) throw new BadRequest('range query parameter must be inputted.');
+
+		const rangeOfNums = this.parseRangeOfNumbersOrThrow(range);
+		const rangeOfNumsToHash = await Promise.all(
+			rangeOfNums.map(async (n) => await this.getHashForBlock(n.toString()))
+		);
+
+		const eventDocsArg = eventDocs === 'true';
+		const extrinsicDocsArg = extrinsicDocs === 'true';
+		const queryFinalizedHead = !this.options.finalizes ? false : true;
+		const omitFinalizedTag = !this.options.finalizes ? true : false;
+		const options = {
+			eventDocs: eventDocsArg,
+			extrinsicDocs: extrinsicDocsArg,
+			checkFinalized: false,
+			queryFinalizedHead,
+			omitFinalizedTag,
+		};
+
+		/**
+		 * Given a range of hashes, generate an array containing callbacks for
+		 * each query.
+		 */
+		const tasks: Array<() => Promise<IBlock>> = [];
+		for (let i = 0; i < rangeOfNumsToHash.length; i++) {
+			tasks.push(async () => {
+				const historicApi = await this.api.at(rangeOfNumsToHash[i]);
+				return await this.service.fetchBlock(
+					rangeOfNumsToHash[i],
+					historicApi,
+					options
+				);
+			});
+		}
+
+		const blocksPromise: Promise<unknown>[] = [];
+
+		const pQueue = new PromiseQueue(4);
+		for (let i = 0; i < tasks.length; i++) {
```

Comment:
does iterators work here? would be nice to get rid of the c-style for loops if possible.